src/Documentation/grains/transactions.md

```diff
@@ -3,82 +3,110 @@ layout: page
+
+Calls can be marked as “Create”, meaning the call will always start its own transaction.
 
 ``` csharp
 public interface IATMGrain : IGrainWithIntegerKey
 {
-    [Transaction(TransactionOption.RequiresNew)]
+    [Transaction(TransactionOption.Create)]
     Task Transfer(Guid fromAccount, Guid toAccount, uint amountToTransfer);
 }
 ```
 
 The Transfer operation in the above atm grain will always start a new transaction which involves the two referenced accounts.
+The transactional operations Withdraw and Deposit on the account grain are marked “Join”, indicating that they can only be called within the context of an existing transaction, which would be the case if they were called during `IATMGrain.Transfer(…)`.
+The `GetBalance` call is marked `CreateOrJoin` so it can be called from within an existing transaction, like via `IATMGrain.Transfer(…)`, or on its own.
 
 ``` csharp
 public interface IAccountGrain : IGrainWithGuidKey
 {
-    [Transaction(TransactionOption.Required)]
+    [Transaction(TransactionOption.Join)]
     Task Withdraw(uint amount);
 
-    [Transaction(TransactionOption.Required)]
+    [Transaction(TransactionOption.Join)]
     Task Deposit(uint amount);
 
-    [Transaction(TransactionOption.Required)]
+    [Transaction(TransactionOption.CreateOrJoin)]
     Task<uint> GetBalance();
 }
 ```
 
-The Withdraw and Deposit operations in the above account grain can take part in transactional operations, like the Transfer operation in the ATM grain, with other transactional grains.
+### Grain Implementations
 
-To use a transactional state within a grain, one needs only define a serializable state class to be persisted and declare the state in the grains constructor.
+A grain implementation needs to use an `ITransactionalState` facet (see Facet System) to manage grain state via ACID transactions.
+
+```csharp
+    public interface ITransactionalState<TState>  
+        where TState : class, new()
+    {
+        Task<TResult> PerformRead<TResult>(Func<TState, TResult> readFunction);
+        Task<TResult> PerformUpdate<TResult>(Func<TState, TResult> updateFunction);
+    }
+```
+
+All read or write access to the persisted state must be performed via synchronous functions passed to the transactional state facet.
+This allows the transaction system to perform or cancel these operations transactionally.
+To use a transactional state within a grain, one needs only define a serializable state class to be persisted and declare the transactional state in the grain’s constructor with a `TransactionalState` attribute which declares the state name, and (optionally) which transactional state storage to use (see Setup).
+
+``` csharp
+    [AttributeUsage(AttributeTargets.Parameter)]
+    public class TransactionalStateAttribute : Attribute
+    {
+        public TransactionalStateAttribute(string stateName, string storageName = null)
+        {
+            …
+        }
+    }
```

Comment:
I believe we should remove the initial indenting here and in any other code blocks

Reply:
I believe this, too.

```csharp

[AttributeUsage(AttributeTargets.Parameter)]
public class TransactionalStateAttribute : Attribute
{
    public TransactionalStateAttribute(string stateName, string storageName = null)
    {
      …
    }
}

```